### ISSUE TITLE
Boolean Integers

### ISSUE THREAD

**Issue by nimylian** [prodis/wannabe_bool#2]
I think that the treatment of integers, in this Gem, is contrary to other places in Ruby. 
0 represents absence, much like nil does. It makes sense that they share the false value.
1 represents presence, as do other values. In conditional statements, the presence of a hash key is truthy. Therefore I propose that non-zero integers should return true. 

Strings that have non-specific values (not "true" or "F") should also be true, because they have value. "F","false", etc are special cases that should be false.

**Comment by thomassnielsen** [comment]
This is how .boolValue in Cocoa behaves as well, probably for the reason you stated. I agree that checking presence is the expected behavior when calling to_b on an integer. 

From the Cocoa docs on boolValue:

> This property is true on encountering one of "Y", "y", "T", "t", or a digit 1-9—the method ignores any trailing characters.

For strings, however, I'm not sure I agree. Getting True on garbage data can potentially be bad, and I've seen cases where the previously mentioned Cocoa behavior (which only check the first character) can cause problems this way.

**Comment by prodis** [comment]
@nimylian You completely right about conversion from integer!
It's fixed in version 0.3.0.

```ruby
require 'wannabe_bool'

0.to_b  # => false
1.to_b  # => true
2.to_b  # => true
-1.to_b # => true
-2.to_b # => true
```

Thank you a lot!
